DATA COLLECTION

In [1]:
import pandas as pd
import numpy as np

In [2]:
df_IDS_ALLCountries_Data = pd.read_csv('IDS_ALLCountries_Data.csv', encoding='latin1')

In [3]:
df_IDS_ALLCountries_Data.shape

(62983, 39)

In [4]:
df_IDS_ALLCountries_Data.columns

Index(['Country Name', 'Country Code', 'Counterpart-Area Name',
       'Counterpart-Area Code', 'Series Name', 'Series Code', '2000', '2001',
       '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010',
       '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2018', '2019',
       '2020', '2021', '2022', '2023', '2024', '2025', '2026', '2027', '2028',
       '2029', '2030', '2031', '2032'],
      dtype='object')

In [5]:
df_IDS_ALLCountries_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62983 entries, 0 to 62982
Data columns (total 39 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Country Name           62980 non-null  object 
 1   Country Code           62978 non-null  object 
 2   Counterpart-Area Name  62978 non-null  object 
 3   Counterpart-Area Code  62978 non-null  object 
 4   Series Name            62978 non-null  object 
 5   Series Code            62978 non-null  object 
 6   2000                   49490 non-null  float64
 7   2001                   49434 non-null  float64
 8   2002                   49262 non-null  float64
 9   2003                   49001 non-null  float64
 10  2004                   49195 non-null  float64
 11  2005                   49137 non-null  float64
 12  2006                   49394 non-null  float64
 13  2007                   49645 non-null  float64
 14  2008                   49497 non-null  float64
 15  20

In [6]:
df_IDS_ALLCountries_Data.head()

,Country Name,Country Code,Counterpart-Area Name,Counterpart-Area Code,Series Name,Series Code,2000,2001,2002,2003,...,2023,2024,2025,2026,2027,2028,2029,2030,2031,2032
0,Afghanistan,AFG,World,WLD,Average grace period on new external debt comm...,DT.GPA.DPPG,NaN,NaN,NaN,NaN,...,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Afghanistan,AFG,World,WLD,Average grace period on new external debt comm...,DT.GPA.OFFT,NaN,NaN,NaN,NaN,...,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Afghanistan,AFG,World,WLD,Average grace period on new external debt comm...,DT.GPA.PRVT,NaN,NaN,NaN,NaN,...,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Afghanistan,AFG,World,WLD,Average grant element on new external debt com...,DT.GRE.DPPG,NaN,NaN,NaN,NaN,...,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Afghanistan,AFG,World,WLD,Average grant element on new external debt com...,DT.GRE.OFFT,NaN,NaN,NaN,NaN,...,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


IDS_ALLCountries_Data - Observations

Contains 62,983 rows and 39 columns
Contains country-level debt indicators
Year columns range from 2000 to 2032
Contains Country Name, Country Code, Counterpart-Area Name, Counterpart-Area code, Series Name, Series Code
Contains missing values in year columns

In [7]:
df_country_series = pd.read_csv('Country-Series - Metadata.csv', encoding='latin1')

In [8]:
df_country_series.shape


(375, 4)

In [9]:
df_country_series.columns

Index(['Type', 'Country Code', 'Series Code', 'Description'], dtype='object')

In [10]:
df_country_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375 entries, 0 to 374
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Type          375 non-null    object
 1   Country Code  375 non-null    object
 2   Series Code   375 non-null    object
 3   Description   375 non-null    object
dtypes: object(4)
memory usage: 11.8+ KB


In [11]:
df_country_series.head()

,Type,Country Code,Series Code,Description
0,Country-Series,Afghanistan (AFG),"Foreign direct investment, net inflows in repo...",Data on Foreign Direct Investment are based on...
1,Country-Series,Afghanistan (AFG),"Population, total (SP.POP.TOTL)",Data source: United Nations World Population P...
2,Country-Series,Afghanistan (AFG),"External debt stocks, total (DOD, current US$)...","Long-term public and publicly guaranteed, long..."
3,Country-Series,Angola (AGO),"External debt stocks, total (DOD, current US$)...",Long-term public and publicly guaranteed debt ...
4,Country-Series,Angola (AGO),"Population, total (SP.POP.TOTL)",Data source: United Nations World Population P...


Country series

Contains 374 rows and 4 columns   

Contains country code, series code, description


In [12]:
df_IDS_CountryMetaData = pd.read_csv('IDS_CountryMetaData.csv', encoding='latin1')

In [13]:
df_IDS_CountryMetaData.shape

(134, 30)

In [14]:
df_IDS_CountryMetaData.columns

Index(['Code', 'Long Name', 'Income Group', 'Region', 'Lending category',
       'Other groups', 'Currency Unit', 'Latest population census',
       'Latest household survey', 'Special Notes',
       'National accounts base year', 'National accounts reference year',
       'System of National Accounts', 'SNA price valuation',
       'PPP survey years', 'Balance of Payments Manual in use',
       'External debt Reporting status', 'System of trade',
       'Government Accounting concept', 'IMF data dissemination standard',
       'Source of most recent Income and expenditure data',
       'Vital registration complete', 'Latest agricultural census',
       'Latest industrial data', 'Latest trade data',
       'Latest water withdrawal data', '2-alpha code', 'WB-2 code',
       'Table Name', 'Short Name'],
      dtype='object')

In [15]:
df_IDS_CountryMetaData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 134 entries, 0 to 133
Data columns (total 30 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Code                                               134 non-null    object 
 1   Long Name                                          134 non-null    object 
 2   Income Group                                       119 non-null    object 
 3   Region                                             120 non-null    object 
 4   Lending category                                   120 non-null    object 
 5   Other groups                                       40 non-null     object 
 6   Currency Unit                                      120 non-null    object 
 7   Latest population census                           118 non-null    object 
 8   Latest household survey                            120 non-null    object 
 9   Special No

In [16]:
df_IDS_CountryMetaData.head()

,Code,Long Name,Income Group,Region,Lending category,Other groups,Currency Unit,Latest population census,Latest household survey,Special Notes,...,Source of most recent Income and expenditure data,Vital registration complete,Latest agricultural census,Latest industrial data,Latest trade data,Latest water withdrawal data,2-alpha code,WB-2 code,Table Name,Short Name
0,AFG,Islamic State of Afghanistan,Low income,Middle East & North Africa,IDA,HIPC,Afghan afghani,1979,Multiple Indicator Cluster Survey 2022-2023,The reporting period for national accounts dat...,...,"Integrated household survey (IHS), 2016/17",NaN,NaN,NaN,2018.0,2000.0,AF,AF,Afghanistan,Afghanistan
1,ALB,Republic of Albania,Upper middle income,Europe & Central Asia,IBRD,NaN,Albanian lek,2023,Demographic and Health Survey 2017 - 2018,NaN,...,Living Standards Measurement Study Survey (LSM...,Yes,2012,2013.0,2018.0,2006.0,AL,AL,Albania,Albania
2,DZA,People's Democratic Republic of Algeria,Upper middle income,Middle East & North Africa,IBRD,NaN,Algerian dinar,2022,Multiple Indicator Cluster Survey 2018-2019,NaN,...,"Integrated household survey (IHS), 2011",NaN,NaN,2010.0,2017.0,2012.0,DZ,DZ,Algeria,Algeria
3,AGO,People's Republic of Angola,Lower middle income,Sub-Saharan Africa,IBRD,NaN,Angolan kwanza,2014,Demographic and Health Survey 2015/16,The World Bank systematically assesses the app...,...,"Integrated household survey (IHS), 2008/09",NaN,NaN,NaN,2018.0,2005.0,AO,AO,Angola,Angola
4,ARG,Argentine Republic,Upper middle income,Latin America & Caribbean,IBRD,NaN,Argentine peso,2022,Multiple Indicator Cluster Survey 2019-2020,The World Bank systematically assesses the app...,...,"Integrated household survey (IHS), 2016",Yes,2008,2002.0,2018.0,2011.0,AR,AR,Argentina,Argentina


IDS_CountryMetaData

Contains 134 rows and 30 columns      

Contains Country Metadata

In [17]:
df_IDS_FootNoteMetaData = pd.read_csv('IDS_FootNoteMetaData.csv', encoding='latin1')

In [18]:
df_IDS_FootNoteMetaData.shape

(2673, 5)

In [19]:
df_IDS_FootNoteMetaData.columns

Index(['Type', 'Country Code', 'Series Code', 'Time Code', 'Description'], dtype='object')

In [20]:
df_IDS_FootNoteMetaData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2673 entries, 0 to 2672
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Type          2673 non-null   object
 1   Country Code  2673 non-null   object
 2   Series Code   2673 non-null   object
 3   Time Code     2673 non-null   object
 4   Description   2673 non-null   object
dtypes: object(5)
memory usage: 104.5+ KB


In [21]:
df_IDS_FootNoteMetaData.head()

,Type,Country Code,Series Code,Time Code,Description
0,FootNote,Afghanistan (AFG),Personal transfers and compensation of employe...,2024 (YR2024),Data on Personal Transfers and Compensation of...
1,FootNote,Afghanistan (AFG),Personal transfers and compensation of employe...,2023 (YR2023),Source: United Nations Conference on Trade and...
2,FootNote,Afghanistan (AFG),Personal transfers and compensation of employe...,2021 (YR2021),Source: United Nations Conference on Trade and...
3,FootNote,Afghanistan (AFG),Personal transfers and compensation of employe...,2022 (YR2022),Source: United Nations Conference on Trade and...
4,FootNote,Afghanistan (AFG),"Foreign direct investment, net inflows in repo...",2021 (YR2021),Source: United Nations Conference on Trade and...


IDS_FootNoteMetaData

Contains 2673 rows and 5 columns



In [22]:
df_IDS_SeriesMetaData = pd.read_csv('IDS_SeriesMetaData.csv',encoding='latin1')

In [23]:
df_IDS_SeriesMetaData.shape

(574, 12)

In [24]:
df_IDS_SeriesMetaData.columns

Index(['Code', 'License Type', 'Indicator Name', 'Short definition',
       'Long definition', 'Source', 'Topic', 'Dataset', 'Periodicity',
       'Aggregation method', 'Limitations and exceptions', 'General comments'],
      dtype='object')

In [25]:
df_IDS_SeriesMetaData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 574 entries, 0 to 573
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Code                        574 non-null    object
 1   License Type                2 non-null      object
 2   Indicator Name              574 non-null    object
 3   Short definition            572 non-null    object
 4   Long definition             574 non-null    object
 5   Source                      574 non-null    object
 6   Topic                       574 non-null    object
 7   Dataset                     573 non-null    object
 8   Periodicity                 574 non-null    object
 9   Aggregation method          574 non-null    object
 10  Limitations and exceptions  1 non-null      object
 11  General comments            8 non-null      object
dtypes: object(12)
memory usage: 53.9+ KB


In [26]:
df_IDS_SeriesMetaData.head()

,Code,License Type,Indicator Name,Short definition,Long definition,Source,Topic,Dataset,Periodicity,Aggregation method,Limitations and exceptions,General comments
0,DT.GPA.DPPG,NaN,Average grace period on new external debt comm...,Grace period is the period from the date of si...,Grace period is the period from the date of si...,"World Bank, International Debt Statistics.",Economic Policy & Debt: External debt: Terms,International Debt Statistics,Annual,Weighted average,NaN,NaN
1,DT.GPA.OFFT,NaN,Average grace period on new external debt comm...,Grace period is the period from the date of si...,Grace period is the period from the date of si...,"World Bank, International Debt Statistics.",Economic Policy & Debt: External debt: Terms,International Debt Statistics,Annual,Weighted average,NaN,NaN
2,DT.GPA.PRVT,NaN,Average grace period on new external debt comm...,Grace period is the period from the date of si...,Grace period is the period from the date of si...,"World Bank, International Debt Statistics.",Economic Policy & Debt: External debt: Terms,International Debt Statistics,Annual,Weighted average,NaN,NaN
3,DT.GRE.DPPG,NaN,Average grant element on new external debt com...,The grant element of a loan is the grant equiv...,The grant element of a loan is the grant equiv...,"World Bank, International Debt Statistics.",Economic Policy & Debt: External debt: Terms,International Debt Statistics,Annual,Weighted average,NaN,NaN
4,DT.GRE.OFFT,NaN,Average grant element on new external debt com...,The grant element of a loan is the grant equiv...,The grant element of a loan is the grant equiv...,"World Bank, International Debt Statistics.",Economic Policy & Debt: External debt: Terms,International Debt Statistics,Annual,Weighted average,NaN,NaN


IDS_SeriesMetaData

Contains 574 rows and 12 columns
